Tests/FluentAssertions.Specs/Specialized/TaskCompletionSourceAssertionSpecs.cs

```diff
@@ -9,8 +8,103 @@ namespace FluentAssertions.Specs.Specialized;
 
 public class TaskCompletionSourceAssertionSpecs
 {
+#if NET6_0_OR_GREATER
+
     [Fact]
     public async Task When_TCS_completes_in_time_it_should_succeed()
+    {
+        // Arrange
+        var subject = new TaskCompletionSource();
+        var timer = new FakeClock();
+
+        // Act
+        Func<Task> action = () => subject.Should(timer).CompleteWithinAsync(1.Seconds());
+        subject.SetResult();
+        timer.Complete();
+
+        // Assert
+        await action.Should().NotThrowAsync();
+    }
+
+    [Fact]
+    public async Task When_TCS_did_not_complete_in_time_it_should_fail()
+    {
+        // Arrange
+        var subject = new TaskCompletionSource();
+        var timer = new FakeClock();
+
+        // Act
+        Func<Task> action = () => subject.Should(timer).CompleteWithinAsync(1.Seconds(), "test {0}", "testArg");
+        timer.Complete();
+
+        // Assert
+        await action.Should().ThrowAsync<XunitException>()
```

Comment:
🤔 I think we should throw a `NullReferenceException` with a clear explanation. This should not happen, unless the test itself is bad.

Reply:
Why you are expecting a NRE?

Reply:
I think, we should not throw NRE explicitly.
https://stackoverflow.com/a/7821803/3621055

Reply:
Sorry, I meant an `ArgumentNullException` using a `Guard.ThrowIfArgumentIsNull`

Reply:
Sorry, now I'm fully out of sync, and the comment does not apply anymore to the current code.
Please explain a little bit more detailed in which situation such guard should be applied.
The new code is a copy of the old class. So you comment applies to both versions?

Reply:
Never mind. I was actually wrong.